assembly/src/parser/lexer.rs

```diff
@@ -609,8 +636,32 @@ fn parse_hex(span: SourceSpan, hex_digits: &str) -> Result<HexEncodedValue, Pars
     }
 }
 
+fn parse_bin(span: SourceSpan, bin_digits: &str) -> Result<BinEncodedValue, ParsingError> {
+    if bin_digits.len() <= 32 {
+        let value =
+            u32::from_str_radix(bin_digits, 2).map_err(|error| ParsingError::InvalidLiteral {
+                span,
+                kind: int_error_kind_to_literal_error_kind(
+                    error.kind(),
+                    LiteralErrorKind::FeltOverflow,
```

Comment:
Is this the right error type here? It's not really "Felt overflow", right?

Reply:
Ah, Indeed! I overlooked it, it should be `U32Overflow`.